move/its/sources/service.move

```diff
@@ -273,20 +287,45 @@ module its::service {
     /// Decode an approved call and check that the source chain is trusted.
     fun decode_approved_message(self: &mut ITS, approved_message: ApprovedMessage): (String, vector<u8>) {
         let (
-            source_chain,
+            mut source_chain,
             _,
             source_address,
-            payload
+            mut payload
         ) = self.channel_mut().consume_approved_message(approved_message);
 
         assert!(self.is_trusted_address(source_chain, source_address), EUntrustedAddress);
 
+        let mut reader = abi::new_reader(payload);
+        if (reader.read_u256() == MESSAGE_TYPE_RECEIVE_FROM_HUB) {
+            assert!(source_chain.into_bytes() == ITS_HUB_CHAIN_NAME, ESenderNotHub);
+            source_chain = ascii::string(reader.read_bytes());
```

Comment:
Add a `read_string` to Abi package that returns a ascii::string for clarity

Reply:
We could do the same for `u64`, `bool`, `vector<String>` and countless other types, where do we draw the line?

Reply:
We should add it for any type that we use. If it makes it cleaner, then why not. Internally, `read_string` can use `read_bytes`, but the caller should avoid dealing with the complexity.